### PR TITLE
Integrate Ovidiu's advanced search

### PR DESCRIPTION
**Pull request by robert-burcea**
Integrated the advanced search functionality from https://github.com/peviitor-ro/ui.orase, created by Ovidiu Alexandru, into the search-engine.

PoC: https://peviitor-rb.web.app/

### CODE REVIEW

File: src/utils/advanced-search.js [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        matchingLocations.push(
+          ...municipiuResult.filter((result) => result !== null)
+        );
+      }
+
+      for (const municipiuLocation of location.municipiu) {
+        const localitateResult = searchLocation(
+          query,
+          municipiuLocation.localitate,
+          location.nume,
```
Comment (OvidiuAl3x) [inline src/utils/advanced-search.js:96]: Schimba location.nume cu  municipiuLocation.nume

File: src/utils/advanced-search.js [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+        matchingLocations.push(
+          ...orasResult.filter((result) => result !== null)
+        );
+      }
+
+      for (const orasLocation of location.oras) {
+        const localitateResult = searchLocation(
+          query,
+          orasLocation.localitate,
+          location.nume,
```
Comment (OvidiuAl3x) [inline src/utils/advanced-search.js:155]: Schimba location.nume cu  orasLocation.nume

File: src/utils/advanced-search.js [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+        matchingLocations.push(
+          ...comunaResult.filter((result) => result !== null)
+        );
+      }
+
+      for (const comunaLocation of location.comuna) {
+        const localitateResult = searchLocation(
+          query,
+          comunaLocation.localitate,
+          location.nume,
```
Comment (OvidiuAl3x) [inline src/utils/advanced-search.js:214]: Schimba location.nume cu  comunaLocation.nume

File: src/utils/advanced-search.js [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+function removeDuplicates(results) {
+  const uniqueResults = [];
+  const seenResults = new Set();
+
+  if (results !== null) {
+    for (const result of results) {
+      const key = `${result.query}-${result.judet}-${result.parent}`;
+
+      if (
+        (result.judet !== result.parent && result.judet !== result.query) ||
```
Comment (OvidiuAl3x) [inline src/utils/advanced-search.js:270]: schimba result.judet !== result.parent && result.judet !== result.query cu result.judet !== result.query && result.query !== result.parent. Comparam doar query cu parent si judet

**Review by sebiboga — APPROVED** [review]

**Pull request merged**